File: src/resources/project-summary.ts
import { ResourceTemplate } from "@modelcontextprotocol/sdk/server/mcp.js";
import type { ToolContext } from "../tools/common.js";

export function registerProjectSummaryResource(server: any, ctx: ToolContext): void {
  server.registerResource(
    "project_summary",
    new ResourceTemplate("projeqtor://projects/{id}/summary", { list: undefined }),
    { title: "Project summary", description: "Consolidated project summary with KPIs source data", mimeType: "application/json" },
    async (uri: URL, vars: any) => {
      const id = Array.isArray(vars.id) ? vars.id[0] : vars.id;
      const data = {
        project: await ctx.client.getObject("Project", id),
        activities: await ctx.client.search("Activity", [{ field: "idProject", operator: "=", value: id }]).catch(() => []),
        tickets: await ctx.client.search("Ticket", [{ field: "idProject", operator: "=", value: id }]).catch(() => []),
        risks: await ctx.client.search("Risk", [{ field: "idProject", operator: "=", value: id }]).catch(() => []),
        issues: await ctx.client.search("Issue", [{ field: "idProject", operator: "=", value: id }]).catch(() => []),
        work: await ctx.client.search("Work", [{ field: "idProject", operator: "=", value: id }]).catch(() => [])
      };
      return { contents: [{ uri: uri.href, mimeType: "application/json", text: JSON.stringify(data, null, 2) }] };
    }
  );
}

File: src/projeqtor_mcp_server/resources/project_summary.py
from __future__ import annotations

import json
from typing import Any

from mcp.server.fastmcp import FastMCP

from src.projeqtor_mcp_server.client.projeqtor_api import ProjeQtOrApiClient, SearchCriterion
from src.projeqtor_mcp_server.tools.common import safe


def register_project_summary_resources(mcp: FastMCP, client: ProjeQtOrApiClient) -> None:
    """Register project summary resource templates."""

    @mcp.resource("projeqtor://projects/{id}/summary")
    async def project_summary(id: str) -> str:
        """Résumé consolidé d'un projet: projet, activités, tickets, risques, issues, work."""
        data: dict[str, Any] = {
            "project": await safe(client.get_object("Project", id)),
            "activities": await safe(client.search("Activity", [SearchCriterion("idProject", id)])),
            "tickets": await safe(client.search("Ticket", [SearchCriterion("idProject", id)])),
            "risks": await safe(client.search("Risk", [SearchCriterion("idProject", id)])),
            "issues": await safe(client.search("Issue", [SearchCriterion("idProject", id)])),
            "work": await safe(client.search("Work", [SearchCriterion("idProject", id)])),
        }
        return json.dumps(data, ensure_ascii=False, indent=2)
